plugins/modules/ec2_snapshot_info.py

```diff
@@ -219,30 +219,38 @@
 from ansible_collections.amazon.aws.plugins.module_utils.ec2 import boto3_tag_list_to_ansible_dict
 
 
-def list_ec2_snapshots(connection, module):
-
-    snapshot_ids = module.params.get("snapshot_ids")
-    owner_ids = [str(owner_id) for owner_id in module.params.get("owner_ids")]
-    restorable_by_user_ids = [str(user_id) for user_id in module.params.get("restorable_by_user_ids")]
-    filters = ansible_dict_to_boto3_filter_list(module.params.get("filters"))
-    max_results = module.params.get('max_results')
-    next_token = module.params.get('next_token_id')
-    optional_param = {}
-    if max_results:
-        optional_param['MaxResults'] = max_results
-    if next_token:
-        optional_param['NextToken'] = next_token
+def build_request_args(snapshot_ids, owner_ids, restorable_by_user_ids, filters, max_results, next_token_id):
 
+    request_args = {
+        'Filters': ansible_dict_to_boto3_filter_list(filters),
+        'MaxResults': max_results,
+        'NextToken': next_token_id,
+        'OwnerIds': owner_ids,
+        'RestorableByUserIds': [str(user_id) for user_id in restorable_by_user_ids],
+        'SnapshotIds': snapshot_ids
+    }
+
+    request_args = {k: v for k, v in request_args.items() if v}
+
+    return request_args
+
+
+def get_snapshots(connection, module, request_args):
+    snapshot_ids = request_args.get("snapshot_ids")
     try:
-        snapshots = connection.describe_snapshots(
-            aws_retry=True,
-            SnapshotIds=snapshot_ids, OwnerIds=owner_ids,
-            RestorableByUserIds=restorable_by_user_ids, Filters=filters,
-            **optional_param)
+        snapshots = connection.describe_snapshots(aws_retry=True, **request_args)
     except is_boto3_error_code('InvalidSnapshot.NotFound') as e:
         if len(snapshot_ids) > 1:
             module.warn("Some of your snapshots may exist, but %s" % str(e))
         snapshots = {'Snapshots': []}
+
+    return snapshots
+
+
+def list_ec2_snapshots(connection, module, request_args):
+
+    try:
+        snapshots = get_snapshots(connection, module, request_args)
```

Comment:
Since `exit_json()` actually call exit() syscall, I would advice to just return a dict and call `module.exit_json()` from outside of the function. This way, to can test the behaviour of your function just by checking the output.

Reply:
Fixed.